src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs

```diff
@@ -561,6 +559,68 @@ public async Task<IActionResult> EditPassword(EditPasswordViewModel model, strin
         return View(model);
     }
 
+    [HttpPost]
+    public async Task<IActionResult> Enable(string id)
+    {
+        if (await _userManager.FindByIdAsync(id) is not User user)
+        {
+            return NotFound();
+        }
+
+        if (!await _authorizationService.AuthorizeAsync(User, UsersPermissions.EditUsers, user))
+        {
+            return Forbid();
+        }
+
+        if (await _userService.EnableAsync(user))
+        {
+            await _notifier.SuccessAsync(H["User account was successfully enabled."]);
+        }
+        else
+        {
+            await _notifier.ErrorAsync(H["Could not enable the user."]);
+        }
+
+        return RedirectToAction(nameof(Index));
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> Disable(string id)
+    {
+        if (await _userManager.FindByIdAsync(id) is not User user)
+        {
+            return NotFound();
+        }
+
+        if (!await _authorizationService.AuthorizeAsync(User, UsersPermissions.EditUsers, user))
+        {
+            return Forbid();
+        }
+
+        var enabledUsersOfAdminRole = (await _userManager.GetUsersInRoleAsync(OrchardCoreConstants.Roles.Administrator))
```

Comment:
This logic should be part of the service. Otherwise we would have to duplicate the same logic in the bulk action and everywhere we use the Display function.